File: OpenVehicleApp/src/com/openvehicles/OVMS/ui/validators/StringValidator.java
package com.openvehicles.OVMS.ui.validators;

import android.text.TextUtils;
import android.widget.EditText;

import com.openvehicles.OVMS.R;

public class StringValidator extends BaseValidator {

	public StringValidator() {
		super();
	}
	
	public StringValidator(String string) {
		super(string);
	}

	@Override
	public boolean valid(EditText pEditText, Object pValue) {
		if (mErrorMessage == null) setErrorMessage(pEditText.getContext().getString(R.string.msg_please_enter_value));
		return !TextUtils.isEmpty((String)pValue);
	}
	
}

File: app/src/main/java/com/openvehicles/OVMS/ui/validators/StringValidator.kt
package com.openvehicles.OVMS.ui.validators

import android.text.TextUtils
import android.widget.EditText
import com.openvehicles.OVMS.R

open class StringValidator(
    override var errorMessage: String? = null
) : Validator {

    override fun valid(editText: EditText?, value: Any?): Boolean {
        if (errorMessage == null) {
            errorMessage = editText!!.context.getString(R.string.msg_please_enter_value)
        }
        return !TextUtils.isEmpty(value as String?)
    }
}
